examples/types/src/lib.rs

```diff
@@ -86,6 +86,12 @@ mod tests {
             Bech32Address::from_str(string).expect("failed to create Bech32 address from string");
         assert_eq!([0u8; 32], *bech32_address.hash());
 
+        // From a hex string.
+        let string = "0x0000000000000000000000000000000000000000000000000000000000000000";
```

Comment:
this is technically not a string but a `&str`, I'm not sure the comment is good here.  Same remark with the ones above, and maybe we could implement an "auto" conversion from `String` -> `bech32_address`.